nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/action/MoveToColdStorageContentAction.java

```diff
@@ -0,0 +1,85 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Guillaume Renard<grenard@nuxeo.com>
+ */
+
+package org.nuxeo.coldstorage.action;
+
+import static org.nuxeo.ecm.core.bulk.BulkServiceImpl.STATUS_STREAM;
+import static org.nuxeo.lib.stream.computation.AbstractComputation.INPUT_1;
+import static org.nuxeo.lib.stream.computation.AbstractComputation.OUTPUT_1;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.coldstorage.service.ColdStorageService;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.bulk.action.computation.AbstractBulkComputation;
+import org.nuxeo.lib.stream.computation.Topology;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.stream.StreamProcessorTopology;
+
+/**
+ * Bulk move documents to Cold Storage.
+ *
+ */
+public class MoveToColdStorageContentAction implements StreamProcessorTopology {
+
+    private static final Logger log = LogManager.getLogger(MoveToColdStorageContentAction.class);
+
+    public static final String ACTION_NAME = "moveToColdStorage";
+
+    @Override
+    public Topology getTopology(Map<String, String> options) {
+        return Topology.builder()
+                       .addComputation(MoveToColdStorageContentComputation::new, //
+                               List.of(INPUT_1 + ":" + ACTION_NAME, OUTPUT_1 + ":" + STATUS_STREAM))
+                       .build();
+    }
+
+    public static class MoveToColdStorageContentComputation extends AbstractBulkComputation {
+
+        public MoveToColdStorageContentComputation() {
+            super(ACTION_NAME);
+        }
+
+        @Override
+        protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
+            log.debug("Start computing documents to be sent to ColdStorage {}", ids);
+            DocumentModelList documents = loadDocuments(session, ids);
+
+            ColdStorageService service = Framework.getService(ColdStorageService.class);
+
+            for (DocumentModel document : documents) {
+                try {
+                    service.moveToColdStorage(session, document.getRef());
+                } catch(NuxeoException e) {
+                    log.warn("Document cannot be moved to Cold Storage: {}",e::getMessage);
```

Comment:
could be interesting to report errors in the status see https://github.com/nuxeo/generator-nuxeo-meta/blob/master/generators/bulk-action/classes/bulk-action.java#L49